pyogrio/_ogr.pyx

```diff
@@ -284,6 +284,33 @@ def _register_drivers():
     GDALAllRegister()
 
 
+cdef void error_handler(CPLErr err_class, int err_no, const char* msg) with gil:
+    """Send CPL debug messages and warnings to Python's logger."""
+
+    # Generally we want to suppress error printing to stderr (behaviour of the
+    # default GDAL error handler) because we already raise a Python exception
+    # that includes the error message
+
+
+    # If the error class is CE_Fatal, we want to have a message issued
+    # because the CPL support code does an abort() before any exception
+    # can be generated
+    if err_class == CPLErr.CE_Fatal:
+        CPLDefaultErrorHandler(err_class, err_no, msg)
+
+    # We do not want to interfere with non-failure messages since
+    # they won't be translated into exceptions.
+    elif err_class != CPLErr.CE_Failure:
+        CPLDefaultErrorHandler(err_class, err_no, msg)
+
+    else:
+        pass
```

Comment:
Still need to clean this up a bit. Probably cleaner to switch the elif/else, and need to clean up the comments (partly copied from GDAL's swig bindings).

We can also check for CE_Warning here, and raise a python warning in that case. But will leave that for a follow-up PR, as that requires some more changes in the tests (to catch the warnings we then generate)

Reply:
Not sure if it helps clean thing up, but conditionals could be combined with return statements - so it is clear that we return as soon as calling the error handler, like:

```Python
    if err_class == CPLErr.CE_Failure:
        # We do not want to interfere with non-failure messages since
        # they won't be translated into exceptions.
        return

    if err_class == CPLErr.CE_Fatal:
        CPLDefaultErrorHandler(err_class, err_no, msg)
        return

    CPLDefaultErrorHandler(err_class, err_no, msg)
```